docs/upgrade/0.7.md

```diff
@@ -0,0 +1,122 @@
+PyQIR 0.7 instead uses only a single package (`pyqir`) that has the functionality of all previous packages.
+
+If you imported the `pyqir.generator` or `pyqir.parser` modules, then the same or an equivalent API is available in the `pyqir` module.
+If you imported the `pyqir.evaluator` module, it is still available under the same name with no API changes.
+
+## Generator
+
+### IR and bitcode conversion
+
+The functions `bitcode_to_ir` and `ir_to_bitcode` were removed because the new `Module` class has the same functionality.
+`Module` supports both parsing and generating QIR.
+For example, instead of:
+
+```python
+from pyqir.generator import bitcode_to_ir, ir_to_bitcode
+
+ir = bitcode_to_ir(bitcode, "module_name")
+bitcode = ir_to_bitcode(ir, "module_name")
+
+# Or with a source filename:
+ir = bitcode_to_ir(bitcode, "module_name", "source_filename")
+```
+
+Use this:
+
+```python
+from pyqir import Context, Module
+
+ir = str(Module.from_bitcode(Context(), bitcode, "name"))
+bitcode = Module.from_ir(Context(), ir, "name").bitcode
+
+# Or with a source filename:
+m = Module.from_bitcode(Context(), bitcode, "module_name")
+m.source_filename = "source_filename"
+ir = str(m)
+```
+
+### Types
+
+If you generated programs with externally-linked functions, then you used the `pyqir.generator.types` module to describe the type of the functions.
+This module has been removed.
+Types need to be created differently because they now directly contain LLVM type objects, which require an LLVM context.
+
+| PyQIR 0.6                                     | PyQIR 0.7                         |
+| --------------------------------------------- | --------------------------------- |
+| `pyqir.generator.types.VOID`                  | `pyqir.Type.void(context)`        |
+| `pyqir.generator.types.BOOL`                  | `pyqir.IntType(context, 1)`       |
+| `pyqir.generator.types.Int(width)`            | `pyqir.IntType(context, width)`   |
+| `pyqir.generator.types.DOUBLE`                | `pyqir.Type.double(context)`      |
+| `pyqir.generator.types.QUBIT`                 | `pyqir.qubit_type(context)`       |
+| `pyqir.generator.types.RESULT`                | `pyqir.result_type(context)`      |
+| `pyqir.generator.types.Function(params, ret)` | `pyqir.FunctionType(ret, params)` |
+
+There are two ways to get a `context` object:
+
+1. Use the `context` property on `SimpleModule`.
+   For example:
+
+   ```python
+   from pyqir import SimpleModule, Type
+   module = SimpleModule("name", num_qubits=1, num_results=1)
+   void = Type.void(module.context)
+   ```
+
+2. Create one yourself.
+   But you also need to give the context you created to `SimpleModule`.
+   For example:
+
+   ```python
+   from pyqir import Context, SimpleModule, Type
+   context = Context()
+   module = SimpleModule("name", num_qubits=1, num_results=1, context=context)
+   void = Type.void(context)
+   ```
+
+## Parser
```

Comment:
I think it would be helpful to have one more section at the end of this that points to the tests in https://github.com/qir-alliance/pyqir/blob/samarsha/upgrade-guide/pyqir/tests/test_parser.py as an example of how the parser is used so folks can see all these patterns in action. Eventually, we should add two new examples to the examples folder: pure parsing and parse, modify, emit. The new examples can be in a follow up PR.

Reply:
I added a section with links to the tests.